src/server/v2.0/handler/robot.go

```diff
@@ -102,6 +103,19 @@ func (rAPI *robotAPI) CreateRobot(ctx context.Context, params operation.CreateRo
 		return rAPI.SendError(ctx, err)
 	}
 
+	existing, err := rAPI.robotCtl.List(ctx, q.New(q.KeyWords{
```

Comment:
I don't think we need to list before creating since the DAO already handle the existing situation.